Tasks/AzureRmWebAppDeployment/operations/PublishProfileUtility.ts

```diff
@@ -0,0 +1,111 @@
+import tl = require('vsts-task-lib/task');
+import { TaskParameters } from './TaskParameters';
+import fs = require('fs');
+import * as Constant from './Constants';
+import path = require('path');
+import Q = require('q');
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var parseString = require('xml2js').parseString;
+const ERROR_FILE_NAME = "error.txt";
+
+export class PublishProfileUtility {
+
+    private _publishProfileJs: any = null;
+    private _publishProfilePath: string;
+
+    constructor(publishProfilePath: string) {
+        this._publishProfilePath = publishProfilePath;
+    }
+
+    public async GetTaskParametersFromPublishProfileFile(taskParams: TaskParameters): Promise<any> {
+        try {
+            if(this._publishProfileJs === null) {
+                this._publishProfileJs = await this.GetPublishProfileJsonFromFile();
+            }
+        } catch(error) {
+            throw new Error(error);
+        }
+        var msDeployPublishingProfile: any = {};
+        taskParams.WebAppName = this._publishProfileJs.DeployIisAppPath[0];
```

Comment:
I would recommend, adding all of these to msdeployPublishingProfile object

Reply:
done